src/nimdowpkg/monitor.nim

```diff
@@ -185,17 +181,14 @@ proc setConfig*(this: Monitor, config: Config) =
 
   for i, tag in this.tags:
     let tagSetting = this.monitorSettings.tagSettings[i + 1]
-    tag.layout.gapSize = this.monitorSettings.layoutSettings.gapSize
-    tag.layout.borderWidth = this.windowSettings.borderWidth
-    tag.layout.masterSlots = tagSetting.numMasterWindows.uint
-    let masterLayout = cast[MasterStackLayout](tag.layout)
-    masterLayout.outerGap = this.monitorSettings.layoutSettings.outerGap
-    masterLayout.defaultWidth = tagSetting.defaultMasterWidthPercentage
-    masterLayout.setDefaultWidth(this.layoutOffset)
+    tag.layout.updateSettings(
+      tagSetting.layoutSettings,
+      this.area,
+      this.windowSettings.borderWidth,
+      this.layoutOffset)
 
   for client in this.taggedClients.clients:
-    if client.borderWidth != 0 or this.monitorSettings.layoutSettings.outerGap > 0:
-      client.borderWidth = this.windowSettings.borderWidth
```

Comment:
This was just done to make things pass. Probably not quite correct, but I'm not entirely sure what this does.